synapse/common.py

```diff
@@ -826,13 +829,38 @@ def setlogging(mlogger, defval=None, structlog=None, log_setup=True, datefmt=Non
 
         log_level = normLogLevel(log_level)
 
+        # See https://docs.python.org/3/howto/logging-cookbook.html#blocking-handlers for info on this configuration
+        logq = queue.Queue()
```

Comment:
```suggestion
        logq = queue.SimpleQueue()
```

Reply:
A queue.Queue has task-done logic tracking; which the QueueListener will detect and utilize in its daemon thread. This is useful if we need to eventually join() on the queue.
But we aren't doing that during atexit.  Instead we're calling the QueueListener.stop() method which poisons the queue and lets the daemon thread clear the entries from the queue until it hits the poison and the thread exits.
Since we aren't join()'ing on the log queue, we could use a queue.SimpleQueue instead.